File: frontend/src/services/users-service.ts
import { apiRequest } from "./api-client";

export function createTeacher(data: {
  name: string;
  email: string;
  department: string;
  employeeId: string;
  designation: string;
  specialization: string;
  password?: string;
  dateOfBirth?: string;
  weddingDate?: string;
}) {
  return apiRequest("/users/teacher", {
    method: "POST",
    body: JSON.stringify(data),
  });
}

export function createStudent(data: any) {
  return apiRequest("/users/student", {
    method: "POST",
    body: JSON.stringify(data),
  });
}

export function getNextStudentIds(): Promise<{ studentId: string; admissionNo: string }> {
  return apiRequest<{ studentId: string; admissionNo: string }>("/users/next-student-ids");
}

export function getAllUsers(): Promise<any[]> {
  return apiRequest<any[]>("/users", {
    headers: {
      "Cache-Control": "no-cache, no-store, must-revalidate",
      "Pragma": "no-cache",
      "Expires": "0",
    },
  });
}

export function deleteUser(id: string) {
  return apiRequest(`/users/${id}`, {
    method: "DELETE",
  });
}

export function updateUserStatus(id: string, status: string) {
  return apiRequest(`/users/${id}/status`, {
    method: "PATCH",
    body: JSON.stringify({ status }),
  });
}

export function bulkUpdateStudentClass(userIds: string[], newClass: string, newSection: string) {
  return apiRequest(`/users/students/bulk-update`, {
    method: "PATCH",
    body: JSON.stringify({ userIds, newClass, newSection }),
  });
}

File: mobile_flutter/lib_yukesh_bak/core/services/users_service.dart
import 'api_service.dart';

class DashboardUser {
  final String id;
  final String name;
  final String email;
  final String role;
  final String department;
  final String status;

  DashboardUser({required this.id, required this.name, required this.email, required this.role, required this.department, required this.status});

  factory DashboardUser.fromJson(Map<String, dynamic> j) => DashboardUser(
    id: j['id']?.toString() ?? '',
    name: j['name'] ?? '',
    email: j['email'] ?? '',
    role: j['role'] ?? '',
    department: j['department'] ?? '',
    status: j['status'] ?? '',
  );
}

class UsersService {
  final ApiService _api;
  UsersService(this._api);

  Future<List<DashboardUser>> getUsers() async {
    final res = await _api.get('/users');
    final data = res.data;
    if (data is List) return data.map((e) => DashboardUser.fromJson(e)).toList();
    return [];
  }

  Future<void> createTeacher(Map<String, dynamic> data) async {
    await _api.post('/users/teacher', data: data);
  }

  Future<void> createStudent(Map<String, dynamic> data) async {
    await _api.post('/users/student', data: data);
  }

  Future<void> deleteUser(String id) async {
    await _api.delete('/users/$id');
  }
}
